Fix/issue 97 fix future date test from articles post file

kat-kan opened the pull request:
## Resolves #97 

### Scope of changes:

Started as fix, evolved into testing more date formats.
This is a draft as I want Your opinion if You like this idea.

### How to use it:

N/A

## Any other comments?

...

Before submitting Pull Request, I've ensured that:

- [x] The PR is associated with the relevant Issue before its creation.
- [x] All new and existing tests passed
- [x] My commit messages follow the `kawqa-gad-playwright` project's [git commit message format](https://github.com/kat-kan/kawqa-gad-playwright/blob/main/CONTRIBUTING.md).

Justyna-KO commented:
@kat-kan 
I think it's a good idea to perform our tets for different dates. 
I like that we don't duplicate the code for these tests thanks to the loop. 
One test fails with me: 
Returns 201 Created after creating article with date 2023-06-24T13:05:45.155Z
Expected: 201
Received: 422
There is a problem with the appropriate assertions for different dates.

Slawomir-DKl reviewed (changes requested):
@kat-kan Great changes! BTW, acc. to https://github.com/jaktestowac/gad-gui-api-demo/releases/tag/v2.6.6, the new version brings fixes in validation of article creation date. The fixes cause both tests "Returns 201 Created after creating article with date" to fail (the tests pass in 2.6.5). I'll try to investigate this today evening and maybe ask Krzysztof about this.

Slawomir-DKl commented on tests/api/articles/articles-post.spec.ts:
```diff
@@ -2,14 +2,22 @@ import { HttpStatusCode } from '@_src_api/enums/api-status-code.enum';
 
 test.describe('POST articles tests', async () => {
   const articles: string = `/api/articles`;
   const articleTitle: string =
```
I've found the reason for the bad result of the "Returns 201 Created..." test. GAD 2.6.6 brings the validation of article title uniqueness - see https://github.com/jaktestowac/gad-gui-api-demo/releases/tag/v2.6.6. Therefore the first run gives one failed test (first POST is OK, 2nd post gives the response
`info: { error: { message: 'Field "title" is not unique!' } }`
I see two solutions:
- restoring DB after each test (which can be troublesome and, in my opinion, is not optimal)
- adding sth like `randomNumber: string = Math.random().toString()` to `articleTitle` variable (note: it must be added also inside the test - due to the loop it is not enough to add the randomization to `test.describe`)

What do you think?

kat-kan replied:
As discussed in the last meeting, both ways are fine and another one is using Faker. No matter which one - I'd prefer to address it in another PR (makes repository history cleaner and more readable later)

Slawomir-DKl commented:
> @kat-kan I think it's a good idea to perform our tets for different dates. I like that we don't duplicate the code for these tests thanks to the loop. One test fails with me: Returns 201 Created after creating article with date 2023-06-24T13:05:45.155Z Expected: 201 Received: 422 There is a problem with the appropriate assertions for different dates.

@Justyna-KO @kat-kan I've found the reason. Please take look at the comment to the line 16

kat-kan commented on tests/api/articles/articles-post.spec.ts:
```diff
@@ -121,27 +133,27 @@ test.describe('POST articles tests', async () => {
+      request,
+    }) => {
+      //Given
+      const futureDateFormatted: string = invalidDate;
```
You are right, I removed it

kat-kan commented:
@Justyna-KO @Slawomir-DKl and any others who'd like to contribute in code review:
as the idea to run tests against different data formats is approved, I removed draft type from PR. I believe it is ready to merge. I added Your suggestions. 
The tests parameterization is yet to be introduced in comments tests - I will create another issue for this task.
I will also create issue for unique article titles (new validation in GAD that was mentioned by @Slawomir-DKl)
